scripts/dev/setup_kind_cluster.sh

```diff
@@ -72,10 +72,26 @@ networking:
   serviceSubnet: "${service_network}"
 containerdConfigPatches:
 - |-
-  [plugins."io.containerd.grpc.v1.cri".registry.mirrors."localhost:${reg_port}"]
-    endpoint = ["http://${reg_name}:${reg_port}"]
+  [plugins."io.containerd.grpc.v1.cri".registry]
+    config_path = "/etc/containerd/certs.d"
```

Comment:
can you explain to me this change and why this is necessary? In meko we are not doing this and it works fine

Reply:
I personally need more information why we are doing this and what it does before approving. I won't block the PR but if it can wait I prefer to let this one open until mircea is back

Reply:
This was basically updating the local registry setup using the official docs guide: https://kind.sigs.k8s.io/docs/user/local-registry/ . This is better because it's not proxying the registry it actually makes the kind node aware of the localhost registry.